src/main.rs

```diff
@@ -346,128 +320,111 @@ enum NodeError {
 /// Return Ok(()) if the channel to the database was closed.
 #[allow(clippy::too_many_arguments)]
 async fn use_node(
-    node_ep: endpoints::Endpoint,
-    token: &str,
+    node_ep: v2::Endpoint,
     sender: &tokio::sync::mpsc::Sender<TransactionLogData>,
     height: &mut AbsoluteBlockHeight, // start height
     max_parallel: u32,
     stop_flag: &AtomicBool,
     canonical_cache: &mut HashSet<AccountAddressEq>,
     max_behind: u32, // maximum number of seconds a node can be behind before it is deemed "behind"
 ) -> Result<(), NodeError> {
-    let mut node =
-        endpoints::Client::connect(node_ep, token).await.map_err(NodeError::ConnectionError)?;
+    let mut node = v2::Client::new(node_ep).await.map_err(NodeError::ConnectionError)?;
     // if the cache is empty we seed it with all accounts at the last finalized
     // block. This will only happen the first time this function is successfully
     // invoked.
     if canonical_cache.is_empty() {
-        let info = node.get_consensus_status().await.context("Error querying consensus status.")?;
-        let accounts = node
-            .get_account_list(&info.last_finalized_block)
+        let accounts_response = node
+            .get_account_list(&v2::BlockIdentifier::LastFinal)
             .await
             .context("Error querying account list.")?;
-        // this relies on the fact that get_account_list returns canonical addresses.
-        log::debug!("Initializing the address cache with {} accounts.", accounts.len());
-        for addr in accounts {
-            canonical_cache.insert(AccountAddressEq(addr));
-        }
+        let accounts = accounts_response.response;
+        let r = accounts
+            .try_fold(HashSet::new(), |mut cc, addr| async move {
+                let _ = cc.insert(AccountAddressEq(addr));
+                Ok(cc)
+            })
+            .await?;
+        *canonical_cache = r;
     }
-    let mut parallel = 1;
+    let mut stream = node.get_finalized_blocks_from(*height).await?;
```

Comment:
Consider provider a more detailed name instead of the type.
maybe `finalized_blocks`